pkg/summary/summarizers.go

```diff
@@ -61,25 +69,40 @@ var (
 		"Processed":                   "processed",
 	}
 
-	// True == error
-	// False ==
-	// Unknown ==
-	ErrorTrue = map[string]bool{
-		"OutOfDisk":           true,
-		"MemoryPressure":      true,
-		"DiskPressure":        true,
-		"NetworkUnavailable":  true,
-		"KernelHasNoDeadlock": true,
-		"Unschedulable":       true,
-		"ReplicaFailure":      true,
-		"Stalled":             true,
-	}
-
-	// True ==
-	// False == error
-	// Unknown ==
-	ErrorFalse = map[string]bool{
-		"Failed": true,
+	// For given GVK, This condition Type and this Status, indicates an error or not
+	// e.g.: GVK: helm.cattle.io/v1, HelmChart
+	//		--> JobCreated: [], indicates True or False are not errors
+	//		--> Failed: ["True"], indicates "True" status is considered error
+	//		--> Worked: ["False"], indicates "False" status is considered error
+	//		--> Unknown: ["True", "False"] indicated "True" or "False" are considered errors
+	GVKConditionErrorMapping = ConditionTypeStatusErrorMapping{
+		{Group: "helm.cattle.io", Version: "v1", Kind: "HelmChart"}: {
+			"JobCreated": sets.New[metav1.ConditionStatus](),
+			"Failed":     sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		},
+		{Group: "", Version: "v1", Kind: "Node"}: {
+			"MemoryPressure":     sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+			"DiskPressure":       sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+			"NetworkUnavailable": sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		},
+		{Group: "apps", Version: "v1", Kind: "Deployment"}: {
+			"ReplicaFailure": sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+			"Stalled":        sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+			"Progressing":    sets.New[metav1.ConditionStatus](metav1.ConditionFalse),
+		},
+	}
+
+	// In case we cannot match any Groups, Versions and Kinds in GVKConditionErrorMapping then we fallback to this
+	// mapping.
+	FallbackConditionErrorMapping = map[string]sets.Set[metav1.ConditionStatus]{
+		"OutOfDisk":          sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		"MemoryPressure":     sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		"DiskPressure":       sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		"NetworkUnavailable": sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		"Unschedulable":      sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		"ReplicaFailure":     sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		"Stalled":            sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
+		"Failed":             sets.New[metav1.ConditionStatus](metav1.ConditionTrue),
```

Comment:
This should only contain Stalled, as all the other condition types are expected to be found on specific Kinds. Stalled can be found on any CR Kind managed via a Rancher controller that uses the rancher kstatus helpers:
https://github.com/rancher/wrangler/blob/master/pkg/kstatus/kstatus.go